custom_components/ocpp/api.py

```diff
@@ -464,30 +467,54 @@ async def handle_set_charge_rate(call):
+                DEFAULT_QUIRK_MONITORED_TO_CHARGER,
+            )
+
             key = ckey.meter_values_sampled_data.value
-            try:
-                chgr_measurands = await self.get_configuration(key)
-            except Exception:
-                _LOGGER.debug(
-                    f"'{self.id}' had error while returning measurands, ignoring"
+
+            # Quirk:
+            # Workaround for a bug on chargers that have invalid MeterValuesSampledData
+            # configuration. This quirk allows to set it to the user provided list.
+            # Corresponding issue: https://github.com/lbbrhzn/ocpp/issues/1275
+            if quirk_monitored_to_charger:
+                _LOGGER.info(
+                    f"'{self.id}' quirk: set OCPP Measurands on charger to '{all_measurands}'"
+                )
+                req = call.ChangeConfiguration(key=key, value=all_measurands)
+                resp = await self.call(req)
+                _LOGGER.info(
+                    f"'{self.id}' quirk: disable setting OCPP measurands via the quirk if your charger does not need it anymore"
                 )
-                chgr_measurands = all_measurands
 
-            accepted_measurands = []
-            cfg_ok = [
-                ConfigurationStatus.accepted,
-                ConfigurationStatus.reboot_required,
-            ]
+            chgr_measurands = await self.get_configuration(key)
 
-            for measurand in all_measurands.split(","):
-                _LOGGER.debug(f"'{self.id}' trying measurand: '{measurand}'")
-                req = call.ChangeConfiguration(key=key, value=measurand)
-                resp = await self.call(req)
-                if resp.status in cfg_ok:
-                    _LOGGER.debug(f"'{self.id}' adding measurand: '{measurand}'")
-                    accepted_measurands.append(measurand)
+            if autodetect_measurands:
+                accepted_measurands = []
+                cfg_ok = [
+                    ConfigurationStatus.accepted,
+                    ConfigurationStatus.reboot_required,
+                ]
+
+                for measurand in all_measurands.split(","):
+                    _LOGGER.debug(f"'{self.id}' trying measurand: '{measurand}'")
+                    req = call.ChangeConfiguration(key=key, value=measurand)
+                    resp = await self.call(req)
+                    if resp.status in cfg_ok:
+                        _LOGGER.debug(f"'{self.id}' adding measurand: '{measurand}'")
+                        accepted_measurands.append(measurand)
 
-            accepted_measurands = ",".join(accepted_measurands)
+                accepted_measurands = ",".join(accepted_measurands)
+            else:
+                accepted_measurands = all_measurands
+                _LOGGER.debug(
+                    f"'{self.id}' measurands set manually to {accepted_measurands}"
+                )
 
             if len(accepted_measurands) > 0:
                 _LOGGER.debug(
```

Comment:
I suggest replacing the `await self.configure(key, accepted_measurands)` here with
```
req = call.ChangeConfiguration(key=key, value=all_measurands)
resp = await self.call(req)
```